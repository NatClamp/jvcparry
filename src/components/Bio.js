import React from 'react'
import { Row, Col, Container, Text, Anchor, Button, Div } from "atomize";
import portrait from '../images/portrait.jpg';


const Bio = () => {

  return (
    <Container className='featureContainer'>
      <Row d='flex' flexDir={{ xs: 'column', sm: 'column', md: 'column', lg: 'row' }} justify={{ xs: 'center', sm: 'space-around' }} align='center'>
        <Col size={{ xs: '10', sm: '7' }} d='flex' flexDir='column' justify='center'>
          <Div m={{ y: '20px' }}>
            <Text textSize="heading" textAlign="right">Greetings!</Text>
            <Text textSize={{ xs: 'paragraph', sm: 'subheader' }} textAlign="right" p={{ b: '10px' }}>I'm Joshua (JVC) Parry [he/him]. </Text>
            <Text textSize={{ xs: 'paragraph', sm: 'subheader' }} textAlign="right" p={{ b: '10px' }}>I'm a designer, author, and publisher of TTRPG material based in the UK. I'm a freelancer in numerous capacities, but primarily as an author of TTRPG material. Much of my work is for large companies such as LoreSmyth and Nord Games, though I work for private commission too. </Text>
            <Text textSize={{ xs: 'paragraph', sm: 'subheader' }} textAlign="right">My formal education is in Biosciences and Zoology, and in my spare time I like to play games (obviously), ramble, write, fish, play in bands, and eat fine food!</Text>
          </Div>
          <Div d="flex" justify='flex-end' m={{ b: '20px' }}>
            <Anchor href="/hire-me" textDecoration='none' w='100px' m='0'>
              <Button
                d='inline-block'
                r='0'
                h="2.5rem"
                p={{ x: "1rem" }}
                textSize="body"
                textColor="black"
                hoverTextColor="warning900"
                bg="white"
                hoverBg="grey600"
                border="1px solid"
                borderColor="black"
                hoverBorderColor="warning900"
              >Hire me</Button>
            </Anchor>
          </Div>

        </Col>
        <Col size={{ xs: '10', sm: '4' }} >
          <img src={portrait} alt='' style={{ maxHeight: '300px' }} />
        </Col>
      </Row>
    </Container>
  )
}

export default Bio
