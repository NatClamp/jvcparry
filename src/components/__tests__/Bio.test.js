/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import Bio from '../Bio';

describe('Bio component', () => {
  afterEach(cleanup);
  const history = createMemoryHistory();
  it('should render the biography text', () => {
    render(
      <Router history={history}>
        <Bio />
      </Router>,
    );
    expect(screen.getByText(/Greetings!/i)).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <Bio />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
