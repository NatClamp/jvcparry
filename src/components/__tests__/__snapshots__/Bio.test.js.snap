// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio component should match the snapshot 1`] = `
<div
  className="bioContainer "
>
  <div
    className=""
  >
    <div
      className=""
    >
      <div
        className=""
      >
        <p
          className=""
        >
          Greetings!
        </p>
        <p
          className=""
        >
          I'm Joshua (JVC) Parry [he/him]. 
        </p>
        <p
          className=""
        >
          I'm a designer, author, and publisher of TTRPG material based in the UK. I'm a freelancer in numerous capacities, but primarily as an author of TTRPG material. Much of my work is for large companies such as LoreSmyth and Nord Games, though I work for private commission too. 
        </p>
        <p
          className=""
        >
          My formal education is in Biosciences and Zoology, and in my spare time I like to play games (obviously), ramble, write, fish, play in bands, and eat fine food!
        </p>
      </div>
      <div
        className=""
      >
        <a
          href="/hire-me"
          onClick={[Function]}
          style={
            Object {
              "textDecoration": "none",
            }
          }
        >
          <button
            className=""
          >
            Hire me
          </button>
        </a>
      </div>
    </div>
    <div
      className=""
    >
      <img
        alt="Portrait of JVC Parry"
        src={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "width": "100% ",
          }
        }
      />
    </div>
  </div>
</div>
`;
